mhs/common/mhs_common/state/work_description.py

```diff
@@ -45,6 +45,24 @@ class EmptyWorkDescriptionError(RuntimeError):
     pass
 
 
+async def update_status_with_retries(wdo: WorkDescription,
```

Comment:
I think we need tests for this method in `test_work_description.py`, to ensure we make the correct number of retries, etc.